vyper/codegen/stmt.py

```diff
@@ -143,6 +144,15 @@ def parse_Call(self):
             funcname = self.stmt.func.id
             return STMT_DISPATCH_TABLE[funcname].build_LLL(self.stmt, self.context)
 
+        elif isinstance(self.stmt.func, vy_ast.Attribute) and self.stmt.func.attr == "append":
+            darray = Expr(self.stmt.func.value, self.context).lll_node
+            args = [Expr(x, self.context).lll_node for x in self.stmt.args]
+            assert len(args) == 1
+            arg = args[0]
+            assert isinstance(darray.typ, DArrayType)
+            assert arg.typ == darray.typ.subtype
```

Comment:
Are `assert` statements invariants (e.g. should never hit) or are they meant to raise actual compiler errors?

Reply:
invariants